File: app/src/main/java/io/githup/limvot/mangagaga/DownloadedActivity.scala
package io.githup.limvot.mangagaga;

import org.scaloid.common._

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ListView;

class DownloadedActivity extends SActivity {
  onCreate {
    contentView = new SVerticalLayout {
        val mangaListView = SListView().<<.wrap.>>
        

        mangaListView.setAdapter(new SArrayAdapter(MangaManager.getSavedManga().toArray))
        mangaListView.setOnItemClickListener( new AdapterView.OnItemClickListener() {
          override def onItemClick(adapterView: AdapterView[_], view: View, i: Int, l: Long) {
            Log.i("onItemClick", mangaListView.getItemAtPosition(i).toString())
            MangaManager.readingOffline(true)
            MangaManager.setCurrentManga(mangaListView.getItemAtPosition(i).asInstanceOf[Manga])
            startActivity[ChapterActivity]
          }
        })
    }
  }
}

File: app/src/main/java/io/githup/limvot/mangagaga/DownloadedActivity.kt
package io.githup.limvot.mangagaga

import org.jetbrains.anko.*
import org.jetbrains.anko.sdk25.coroutines.onClick

import android.os.Bundle
import android.app.Activity

class DownloadedActivity : Activity(), GenericLogger {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        verticalLayout {
            listView {
                val listItems = Boss.getSavedManga().map { req ->
                                                TextListItem(req.manga, {
                                                    Boss.currentSource = req.source // this is "downloaded"
                                                    Boss.currentManga = req.manga
                                                    startActivity<ChapterActivity>()
                                                }) }
                adapter = SimpleListAdaptor(ctx, listItems)
            }
        }
    }
}
